sherpa/plot/backends.py

```diff
@@ -774,6 +807,20 @@ def as_html_contour2d(self, plot):
                              'min', 'max', 'nloop',
                              'delv', 'fac', 'log'])
 
+    # TODO: Inconsistent. Some plots have as_html_xxx, but others do not.
```

Comment:
As I've mentioned before, I do wonder whether we should just have a generic `get_as_html` type routine which takes the name of the "plot type" - e.g. "histogram", "image", ... - rather than having multiple `as_html_xxx` routines (naming is one of the issues here as we already have a `as_html` method with a slighty-different API if memory serves me.

But this is all for a different PR than this one.

Reply:
> As I've mentioned before, I do wonder whether we should just have a generic get_as_html ...
I've suggested one approach in #1375 to this, but that's for a different PR as you say.